packages/framework/src/Model/Order/Order.php

```diff
@@ -938,260 +927,15 @@ public function isCancelled()
-     * @param \Shopsys\FrameworkBundle\Model\Order\Item\OrderItemFactoryInterface $orderItemFactory
-     * @param \Shopsys\FrameworkBundle\Model\Pricing\Price $productsPrice
-     * @param string $locale
-     */
-    public function fillOrderPayment(
-        PaymentPriceCalculation $paymentPriceCalculation,
-        OrderItemFactoryInterface $orderItemFactory,
-        Price $productsPrice,
-        $locale
-    ) {
-        $payment = $this->getPayment();
-        $paymentPrice = $paymentPriceCalculation->calculatePrice(
-            $payment,
-            $this->getCurrency(),
-            $productsPrice,
-            $this->getDomainId()
-        );
-        $orderPayment = $orderItemFactory->createPayment(
-            $this,
-            $payment->getName($locale),
-            $paymentPrice,
-            $payment->getVat()->getPercent(),
-            1,
-            $payment
-        );
-        $this->addItem($orderPayment);
-    }
-
-    /**
-     * @param \Shopsys\FrameworkBundle\Model\Transport\TransportPriceCalculation $transportPriceCalculation
-     * @param \Shopsys\FrameworkBundle\Model\Order\Item\OrderItemFactoryInterface $orderItemFactory
-     * @param \Shopsys\FrameworkBundle\Model\Pricing\Price $productsPrice
-     * @param string $locale
-     */
-    public function fillOrderTransport(
-        TransportPriceCalculation $transportPriceCalculation,
-        OrderItemFactoryInterface $orderItemFactory,
-        Price $productsPrice,
-        $locale
-    ) {
-        $transport = $this->getTransport();
-        $transportPrice = $transportPriceCalculation->calculatePrice(
-            $transport,
-            $this->getCurrency(),
-            $productsPrice,
-            $this->getDomainId()
-        );
-        $orderTransport = $orderItemFactory->createTransport(
-            $this,
-            $transport->getName($locale),
-            $transportPrice,
-            $transport->getVat()->getPercent(),
-            1,
-            $transport
-        );
-        $this->addItem($orderTransport);
-    }
-
-    /**
-     * @param \Shopsys\FrameworkBundle\Model\Order\Preview\OrderPreview $orderPreview
-     * @param \Shopsys\FrameworkBundle\Model\Order\Item\OrderItemFactoryInterface $orderItemFactory
-     * @param \Shopsys\FrameworkBundle\Twig\NumberFormatterExtension $numberFormatterExtension
-     * @param string $locale
-     */
-    public function fillOrderProducts(
-        OrderPreview $orderPreview,
-        OrderItemFactoryInterface $orderItemFactory,
-        NumberFormatterExtension $numberFormatterExtension,
-        $locale
-    ) {
-        $quantifiedItemPrices = $orderPreview->getQuantifiedItemsPrices();
-        $quantifiedItemDiscounts = $orderPreview->getQuantifiedItemsDiscounts();
-
-        foreach ($orderPreview->getQuantifiedProducts() as $index => $quantifiedProduct) {
-            $product = $quantifiedProduct->getProduct();
-            if (!$product instanceof Product) {
```

Comment:
this wasn't transfered, why?

Reply:
it is explained in the commit message :slightly_smiling_face:
```
- it is not possible to create QuantifiedProduct without Product instance
- if $product was null, then get_class($product) would return unexpected result would return an unexpected result

```
the original motivation is [this comment](https://github.com/shopsys/shopsys/pull/1123#discussion_r292283440)